packages/test-runner/src/workerRunner.ts

```diff
@@ -52,20 +54,28 @@ export class WorkerRunner extends EventEmitter {
   private _parametersString: string;
   private _workerIndex: number;
 
-  constructor(entry: TestRunnerEntry, config: Config, workerIndex: number) {
+  constructor(entry: RunPayload, config: Config, workerIndex: number) {
     super();
     this._suite = new WorkerSuite('');
     this._suite.file = entry.file;
     this._workerIndex = workerIndex;
     this._parametersString = entry.parametersString;
-    this._ids = new Set(entry.ids);
-    this._remaining = new Set(entry.ids);
+    this._entries = new Map(entry.entries.map(e => [ e.testId, e ]));
+    this._remaining = new Map(entry.entries.map(e => [ e.testId, e ]));
     this._trialRun = config.trialRun;
     this._config = config;
     for (const {name, value} of entry.parameters)
       this._parsedParameters[name] = value;
     this._parsedParameters['config'] = config;
     this._parsedParameters['workerIndex'] = workerIndex;
+    let isRetryRun: boolean | undefined;
+    for (const entry of this._entries.values()) {
+      if (isRetryRun === undefined)
+        isRetryRun = entry.retryNumber > 0;
+      else
+        assert(isRetryRun === (entry.retryNumber > 0));
```

Comment:
I'm not sure what you are asserting here. If we don't have isRetryRun set, then we set it, otherwise we assert that all of them have retryNumber?